src/llmcompressor/transformers/finetune/data/base.py

```diff
@@ -83,21 +82,85 @@ def __init__(
             self.max_seq_length = None
             self.padding = False
 
-    def get_raw_dataset(self, cache_dir: Optional[str] = None) -> Dataset:
+    def __call__(self, add_labels: bool = True) -> DatasetType:
+        dataset = self.data_args.dataset
+
+        if isinstance(dataset, str):
+            # load dataset: load from huggingface or disk
+            dataset = self.load_dataset()
```

Comment:
Do we need to abstract this away as opposed to letting the user load the dataset?

Reply:
The user is allowed to load and pass their own custom `Dataset`. This line only executes if the `dataset` argument is a string not a `Dataset` loaded by the user.